docs/src/literate/hyperelasticity.jl

```diff
@@ -18,36 +18,42 @@
 # In this example we will solve a problem in a finite strain setting using an
 # hyperelastic material model. In order to compute the stress we will use automatic
 # differentiation, to solve the non-linear system we use Newton's
-# method, and for solving the Newton increment we use conjugate gradient.
+# method, and for solving the Newton increment we use conjugate gradients.
 #
-# The weak format is expressed in terms of the first Piola-Kirchoff stress ``\mathbf{P}``
+# The weak form is expressed in terms of the first Piola-Kirchoff stress ``\mathbf{P}``
 # as follows: Find ``u \in \mathbb{U}`` such that
 #
 # ```math
-#  \int_\Omega [\delta \mathbf{u} \otimes \nabla] : \mathbf{P}(\mathbf{u})\ \mathrm{d}\Omega =
+# \int_\Omega \nabla \delta \mathbf{u} : \mathbf{P}(\mathbf{u})\ \mathrm{d}\Omega =
```

Comment:
but then it should rather be `Grad` instead of `grad`, shouldn't it?

Reply:
Thanks for the feedback! Maxi is right, it should be `Grad` then - which is why I decided against this notation, trying to avoid confusion.

Regarding `\delta \mathbf{u} \otimes \nabla`, this notation is super problematic, because we change the "direction" of the operator acting. Making a pathological example, what would be `\delta \mathbf{u} \otimes \nabla \otimes \delta \mathbf{v}`? From a pure technical side "\otimes" (and also other products) is not commutative anymore when we involve these kinds of operators. It just happens that some groups define `\delta \mathbf{u} \otimes \nabla := \nabla \otimes \delta \mathbf{u}` - which is fine, but super confusing. I can rewrite it as `\nabla \otimes \delta \mathbf{u}`, if you like this one better. :) 

To give a more realistic example which almost always leads to confusion: Take the Navier-Stokes equations. They have `(v \cdot \nabla) v` and they have, in the incompressible case, `\nabla \cdot v = 0`. However, this does not make the term `(v \cdot \nabla) v` zero, because the inner product is not communative on this space. 

Does this make sense for you?

Reply:
Sure, I think `Grad` is just another way of specifying wrt. to initial coordinates (as opposed to subscript 0 or X)? As long as we write that, I think that is fine and the most safe option.

Regarding the issues with using the open product, I agree that this is problematic. But when groups use `\nabla \otimes \delta \mathbf{u}`, this only works for symmetric tensors. To ensure both correct order wrt differentiation and which tensor leg it acts on, I guess it should be `grad(u)=[\nabla \otimes u]^T`. 
Regarding you example from NS, I would write that as `v \cdot grad(v) = v \cdot [v \otimes \nabla]` if I understand the notation correctly?
(Just so we are not talking past each other, I define `\nabla = [\partial / \partial x_i] e_i` where `e_i` are the basis vectors, or conversely \nabla_0 with X_i instead. ).

Reply:
>Sure, I think Grad is just another way of specifying wrt. to initial coordinates (as opposed to subscript 0 or X)?

Yes.

>[...] But when groups use \nabla \otimes \delta \mathbf{u}, this only works for symmetric tensors. To ensure both correct order wrt differentiation and which tensor leg it acts on, I guess it should be grad(u)=[\nabla \otimes u]^T.

Definitely correct. And I also think that this is where this notation comes from ("transpose applied").

> Regarding you example from NS, I would write that as v \cdot grad(v) = v \cdot [v \otimes \nabla] if I understand the notation correctly?

`(v \cdot \nabla) v` becomes `v_i \partial_i v_j e_j` in index notation while `v \cdot grad(v)` becomes `v_i e_i \cdot (\partial_j e_j \otimes v_k e_k)^T = v_i v_i \partial_j e_j`. So it should be  rather`v \cdot grad(v)^T` I guess.